std/algorithm/sorting.d

```diff
@@ -3753,57 +3753,45 @@ void topNIndex(alias less = "a < b", SwapStrategy ss = SwapStrategy.unstable,
 if (isRandomAccessRange!Range &&
     isRandomAccessRange!RangeIndex &&
     hasAssignableElements!RangeIndex &&
-    isIntegral!(ElementType!(RangeIndex)))
+        isIntegral!(ElementType!(RangeIndex)) || is(ElementType!(RangeIndex) == ElementType!(Range)*))
```

Comment:
The next, to be grouped overload is
 ```d
  isRandomAccessRange!Range &&
  isRandomAccessRange!RangeIndex &&    
  hasAssignableElements!RangeIndex &&
  is(ElementType!(RangeIndex) == ElementType!(Range)*))
```
Notice that the first three constraints are exactly equal and only the last one differs as the first overload is for a range of indexes and the second one for a range of pointers.